### ISSUE TITLE
static user definition in system

### ISSUE THREAD

**Issue by port22** [skavanagh/KeyBox#122]
Hi, 
I don't get it - in a team with 5 administrators (where everyone has an individual key), and say 10 servers, the main use is to let every administrator login with his own username and key combination.

But actually, with keybox one has to define a special username to a server, and then all administrators log in with this same user? makes no sense to me. root is not allowed as login, so the ability to assign servers to users (get login name from keybox-username).
On the other hand, creating a server for every user makes no sense, too.
How is it supposed to be used?

**Comment by skavanagh** [comment]
It can be configured as needed (add the same host for multiple system users).

One of the ideas is to audit outside the servers themselves and audit from one place, even if users have shared accounts on the servers.

You can set this flag to audit inside the app itself
 
https://github.com/skavanagh/KeyBox/blob/master/src/main/resources/KeyBoxConfig.properties#L18

or send the logs to a logging server

https://github.com/skavanagh/KeyBox/blob/master/src/main/resources/log4j.xml#L13-L25

so you still have per user accountability even if connecting to the host as the same user.

**Comment by port22** [comment]
That is not good.  If you have 100 servers and 10 Users, you must create 10 hosts for every user/server combination. Just because there is a static username field in the host description. 

Second, if everybody logs in with a special account like root, you can not retrace which user did bad things. That's what the /var/log/auth+secure is for. Keybox can not provide this kind of auditing, can you ?

**Comment by skavanagh** [comment]
> That is not good. If you have 100 servers and 10 Users, you must create 10 hosts for every user/server combination. Just because there is a static username field in the host description. 

You configure it how you need it.  Have all your keybox users share the same server user in that case.

> Second, if everybody logs in with a special account like root, you can not retrace which user did bad things.  

If you turn auditing on, you still have per user audibility, even if the keybox users are connecting to the host as the same user (even as root), It saves everything.

**Comment by port22** [comment]
> Have all your keybox users share the same server user in that case.

but that is not possible. every user HAS to have his own username for auditing reasons.

> It saves everything

you mean it logs all commands executed by a user on a machine?

Anyway it's very untypical to have a shared user for different people, not at least because every one has it's own bash_profile, cronjobs,  etc. 

Keybox has to have the possibility to share ONE host definition with multiple users, taking the login user from an additional username field in the keybox-user-properties. 

Wouldn't that be possible?

**Comment by skavanagh** [comment]
> you mean it logs all commands executed by a user on a machine?

Yes - if you have auditing turned on it stores every command and associates it with the user you login to KeyBox with not the user on the system.

**Comment by colandre** [comment]
I think the misunderstanding here is that the audit mentioned by skavanagh is the one inside KeyBox web portal (or saved somewhere else via log4j configuration with logstash for example).

I think that silvio (portal22) instead is mentioning local audit on systems at OS level (usually under /var/log folder), that with single user does not distinguish between different users logged with same system account)

**Comment by port22** [comment]
correct. Even it's a very nice idea to have auditing within keybox, it perhaps does not cover all the features that the usual auditing tools in linux are covering. I think it's best to leave that things to the tools which do it best and maybe collect them afterwards in a central location.

Anyway, sharing a login user is another nice idea, but like said, also this misses features like user-specific configuration like mentioned (bashrc, env, etc). 

For now it seems I have to deal with it, but if you mind, keep that in mind for future releases, I think it would not be too hard to make that function work

> Keybox has to have the possibility to share ONE host definition with multiple users, taking the login user from an additional "username"-field in the keybox-user-properties, for example (or just taking the keybox-login-username). Triggering that if the username field in the host definition is blank)

**Comment by colandre** [comment]
I agree with you @port22, but take in consideration that KeyBox is not an auditing tool. It has been created for other kind of needs and it can also perform some auditing features.

In regard with login on systems with KeyBox username (or asking the username), this would be great. The main problem here in my opinion is the user provisioning procedures on systems.
At the moment you can create a new user in KeyBox, assign profile, and without any other operation he/she is able to connect to systems. For user specific access to systems you also need to create user accounts on Linux boxes.

This is the reason why all (normal) users created on KeyBox have the user type "Administrative Only". All users are owners/responsible of all the systems assigned via profiles, and usually they can perform any kind of operations on systems (for example in my case to switch to root user).

**Comment by vinsf** [comment]
@colandre are you saying it is normal for users to be able to ssh to any host that is in a profile they also belong to, even though the users haven't created a ssh key yet?
